scripts/build_debian_base_system.sh

```diff
@@ -74,7 +76,8 @@ for ((i=0;i<LENGTH;i++))
 do
     package=${PACKAGE_ARR[$i]}
     packagename=$(echo $package | sed -E 's/=[^=]*$//')
-    url=$(echo "$URL_ARR" | grep "/${packagename}_")
+    encoded_packagename=$(urlencode $packagename)
+    url=$(echo "$URL_ARR" | grep -i "/${packagename}_\|/${encoded_packagename}_")
```

Comment:
>grep [](http://example.com/codeflow?start=28&length=4)

Instead of encoding and grep multiple pattern, it is easier to decode `$URL_ARR` using `unquote_plus`, and grep the plaintext pattern.

Reply:
> > grep
> 
> Instead of encoding and grep multiple pattern, it is easier to decode `$URL_ARR` using `unquote_plus`, and grep the plaintext pattern.

Because we need to use the encoded one to download the image. If using the decoded one, we need an additional step to recover back the original one.